pennylane/ops/functions/matrix.py

```diff
@@ -219,10 +219,23 @@ def circuit():
             raise TransformError("Input is not an Operator, tape, QNode, or quantum function")
 
         return _matrix_transform(op, wire_order=wire_order)
-    try:
+

```

Comment:
```suggestion

   try:
       if op.has_matrix:
           return op.matrix(wire_order=wire_order)
       if op.has_sparse_matrix:
           return op.sparse_matrix(wire_order=wire_order).todense()
    except:  # pylint: disable=bare-except
```
For example like this

Reply:
outdated since we found out that the two failed are not related to our modification and have their own issues. Decided to xfail.